plugins/flatpak/gs-flatpak.c

```diff
@@ -3666,6 +3667,51 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static gboolean
+gs_flatpak_allow_install_or_update (GsFlatpak *self,GsApp *app, GsAppList *list, gboolean is_update)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autofree gchar *path = NULL;
+	GsApp *runtime, *update_runtime;
+	guint64 runtime_delta_size = 0;
+	guint64 min_free_space = 0;
+	guint i;
+	struct statvfs buf;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+	path = g_file_get_path (installation_dir);
+
+	if (statvfs (path, &buf) == -1) {
+		g_debug ("statvfs: Couldn't query disk space information");
+		return FALSE;
+	}
+
+	runtime = gs_app_get_runtime (app);
+	update_runtime = gs_app_get_update_runtime (app);
+	if (is_update && runtime != update_runtime) {
+
+		/* Correctly setup the runtime sizes if they are updatable */
+		guint64 runtime_size = gs_app_get_size_installed (runtime);
+		guint64 update_runtime_size = gs_app_get_size_installed (update_runtime);
+
+		if ((runtime_delta_size = update_runtime_size - runtime_size) > update_runtime_size)
```

Comment:
What if the `runtime_size` is actually greater than the update's?

Reply:
Also, `update_runtime_size - runtime_size > update_runtime_size` means `runtime_size < 0`, which shouldn't really happen, meaning this if's body will never be used.

Reply:
`gs_app_get_size_installed` returns `guint64`. This check basically checks for wrapping around. If wrapping around is detected (means `runtime_size` > `update_runtime_size`), then reverse the parameters (the `if`'s body)